src/commands/main.rs

```diff
@@ -75,6 +75,14 @@ command!(game(ctx, message, args) {
 command!(name(ctx, message, args) {
     if ALLOWED_USER_IDS.contains(&message.author.id.0) {
         let arg = args.join(" ");
-        let _ = ctx.edit_profile(|p| p.username(arg.as_str()));
+        let _ = ctx.edit_profile(|p| p.username(arg.as_str())).expect("could not set name");
+    }
+});
+
+command!(nick(ctx, message, args) {
+    let _ = ctx;
+    if ALLOWED_USER_IDS.contains(&message.author.id.0) {
+        let arg = args.join(" ");
+        let _ = message.guild_id().unwrap().edit_nickname(Some(arg.as_str()));
```

Comment:
So `guild_id()` returns `Option<T>` because a message may not necessarily be tied to a guild.
You must ensure that `message` absolutely refers to a guild message, and not one from say a DM.

I don't see any checks or other indications of a guarantee.